src/webui/api/torrentscontroller.cpp

```diff
@@ -887,6 +887,32 @@ void TorrentsController::removeTrackersAction()
         torrent->forceReannounce();
 }
 
+void TorrentsController::removeTrackerAction()
+{
+    requireParams({u"urls"_s});
+
+    const QStringList urls = params()[u"urls"_s].split(u'|', Qt::SkipEmptyParts);
+
+    for (BitTorrent::Torrent *const torrent : asConst(BitTorrent::Session::instance()->torrents()))
+    {
+        const QList<BitTorrent::TrackerEntryStatus> trackers = torrent->trackers();
+        QStringList trackersToRemove;
+        trackersToRemove.reserve(trackers.size());
+        for (const BitTorrent::TrackerEntryStatus &trackerEntryStatus : trackers)
+        {
+            for (const QString &urlStr : urls)
+            {
+                const QString url = QUrl::fromPercentEncoding(urlStr.toUtf8());
```

Comment:
Isn't it possible to avoid decoding each URL multiple times?

Reply:
Yes, I will optimize this.